hotness/buildsys.py

```diff
@@ -152,9 +157,40 @@ def handle(self, package, upstream, version, rhbz):
             # Copy the patch out of this doomed dir so bz can find it
             destination = os.path.join('/var/tmp', filename)
             shutil.move(os.path.join(tmp, filename), destination)
-
             return task_id, destination, '[patch] ' + comment
         finally:
             self.log.debug("Removing %r" % tmp)
             shutil.rmtree(tmp)
             pass
+
+    def rebase_helper(self, package, upstream, tmp, bz):
+        """
+        Rebase helper part which does a rebase a inform package
+        maintainer whether package was rebased properly.
+        Output information are in dictionary rh_stuff.
+
+        """
+        self.log.info("Rebase-helper is going to rebase package")
+        rh_stuff = {}
+        result_rh = -1
+        try:
+            url = self.git_url.format(package=package)
+            self.log.info("Cloning %r to %r" % (url, tmp))
+            sh.git.clone(url, tmp)
+            os.chdir(tmp)
+            self.log.info("Rebasehelper package %s %s" % (package, upstream))
+            argument = ['--non-interactive', '--buildtool', 'fedpkg', upstream]
+            cli = CLI(argument)
+            rh_app = Application(cli)
+            rh_app.set_upstream_monitoring()
+            result_rh = rh_app.run()
```

Comment:
Does this call to ``.run()`` create a koji scratch task and then wait for it to complete?

If so, this could be problematic for the-new-hotness running in production.

The way it worked before was that:

- hotness would receive a message from anitya, try to bump the spec, kick off a scratch build, remember the task id, and then it **returned and went back to its main loop**.
- while koji was working on that scratch task, the-new-hotness would continue processing *other* messages... potentially creating more scratch tasks in the mean time.
- when koji finally finished that original scratch task, it would publish a message, hotness would notice, and then follow up on the bug it filed.

If rebase-helper waits for koji to finish each task, I don't think the-new-hotness will drown in its workload.  It will sit there waiting for koji to finish a task, meanwhile thousands of messages will pile up in its inbox.